__tests__/commands/_helpers.js

```diff
@@ -9,6 +9,7 @@ import {run as check} from '../../src/cli/commands/check.js';
 import * as fs from '../../src/util/fs.js';
 import {Install} from '../../src/cli/commands/install.js';
 import Config from '../../src/config.js';
+import {userHomeDir, setUserHomeDir} from '../../src/util/user-home-dir';
```

Comment:
I think for tests this can be achieved with a simple jest mock, https://github.com/yarnpkg/yarn/blob/ed6b9b99c4a83ec9ab8a113892ba900365316b5d/__tests__/commands/install/integration.js#L41

Then you won't need to do setter/getter and change the code